reactioncommerce/reaction
# feat: add refunds GraphQL query

Part of #5158 
Impact: **minor**  
Type: **feature**

A new `listRefunds` query allows you to query refunds of payments on a particular order. This was previously a meteor only method. The meteor method will be removed with #5158.

## Breaking changes
None

## Testing
1. Place an order using both the Stripe payment method and IOU method
1. Create a refund for each type of payment
1. Use the new `listRefunds` query to see the refunds

```
listRefunds(orderId: {ORDER-ID}, shopId: {SHOP-ID}) {
   _id
    amount
    ...otherFieldsYouWant
  }
```

I've added tests here to test the permission / requirements of variables, but did not add tests to check a result, as the payment method has it's own tests for this element: https://github.com/reactioncommerce/reaction/blob/feat-kieckhafer-listRefundsGQLQuery/imports/plugins/included/payments-stripe/server/no-meteor/util/stripeListRefunds.test.js#L10
